server/commands.py

```diff
@@ -617,3 +631,55 @@ def ooc_cmd_mutepm(client, arg):
         client.send_host_message('You stopped receiving PMs')
     else:
         client.send_host_message('You are now receiving PMs')
+
+def ooc_cmd_ban(client, arg):
+    if not client.is_mod:
+        raise ClientError('You must be authorized to do that.')
+    ip = arg.strip()
+    if len(ip) < 7:
+        raise ArgumentError('You must specify a valid IP.')
+    try:
+        client.server.ban_manager.add_ban(ip)
+    except ServerError:
+        raise
+    hdid = client.server.client_manager.get_hdid_by_ip(ip)
+    if hdid == None:
+        client.send_host_message('Unable to locate client HDID for logging.')
+    targets = client.server.client_manager.get_targets_by_ip(ip)
+    if targets:
+        for c in targets:
+            c.disconnect()
```

Comment:
The notice is that you are disconnected. I have no idea how one could display the message that you've been kick/banned within the AO client framework anyway.